20-kubernetes/12-pingdirectory-upgrade/3-staging.yaml

```diff
@@ -0,0 +1,110 @@
+apiVersion: v1
+data:
+  K8S_STATEFUL_SET_NAME: pingdirectory
+  K8S_STATEFUL_SET_SERVICE_NAME: pingdirectory
+  MAKELDIF_USERS: "1000"
+  ORCHESTRATION_TYPE: KUBERNETES
+  PING_IDENTITY_ACCEPT_EULA: "YES"
+  TAIL_LOG_FILES: '""'
+  SERVER_PROFILE_PATH: baseline/pingdirectory
+  SERVER_PROFILE_URL: https://www.github.com/pingidentity/pingidentity-server-profiles.git
+kind: ConfigMap
+metadata:
+  annotations: {}
+  labels:
+    role: pingdirectory
+  name: pingdirectory-environment-variables
+  
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    role: pingdirectory
+  name: pingdirectory  
+spec:
+  clusterIP: None
+  ports:
+  - name: ldaps
+    port: 1636
+  - name: ssl
+    port: 1443
+  - name: ldap
+    port: 1389
+  selector:
+    role: pingdirectory
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  labels:
+    role: pingdirectory
+  name: pingdirectory
+spec:
+  replicas: 2
+  updateStrategy:
+    type: RollingUpdate
+    #this is the important part
+    rollingUpdate:
+      partition: 1
+  selector:
+    matchLabels:
+      role: pingdirectory
+  serviceName: pingdirectory
+  template:
+    metadata:
+      labels:
+        role: pingdirectory
+      name: pingdirectory
+    spec:
+      containers:
+      - envFrom:
+        - configMapRef:
+            name: pingdirectory-environment-variables
+        - secretRef:
+            name: devops-secret
+            optional: true
+        image: pingidentity/pingdirectory:8.1.0.0
```

Comment:
Maybe place a comment before this that it is the important change.  Call it out to it's obvious.